cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -94,18 +98,21 @@ public void validateProperties(Map<String, String> properties) {
     // Lower case letters and dashes allowed only.
     List<String> networkTags = conf.getNetworkTags();
     if (!networkTags.stream().allMatch(e -> NETWORK_TAGS_PATTERN.matcher(e).matches())) {
-      throw new IllegalArgumentException("Invalid Network Tags: Ensure tag length is max 63 chars"
-                                           + " and contains  lowercase letters, numbers and dashes only. ");
+      throw new InvalidConfigurationException("Invalid Network Tags: Ensure tag length is max 63 chars"
+                                           + " and contains  lowercase letters, numbers and dashes only. ",
+              ErrorGroup.CONFIGURATION, ErrorCodeProvider.NETWORK_TAGS);
```

Comment:
would be better for the code to be more specific, that these are invalid network tags (this should be a user error)

Reply:
This is the only place where NETWORK_TAGS is used. The documentation for this error code will mention that the network tags are invalid.

Reply:
There is no point in having error codes if they are not as specific as possible. Documentation cannot help narrow down the cause when the code itself is not narrow enough. There can be other issues with tags in the future, this one should specifically be about invalid characters.

Reply:
Removed error codes based on our discussion. Adding specific error codes for each situation is cumbersome.